File: ios/OlyPoseCameraModule.swift
import ExpoModulesCore

public class OlyPoseCameraModule: Module {
  public func definition() -> ModuleDefinition {
    Name("OlyPoseCamera")
    View(OlyPoseCameraView.self) {
      Events("onCameraConfigured", "onLandmark", "onInferenceError")
      Prop("cameraFacing") { (view: OlyPoseCameraView, facing: String) in view.options.facing = facing }
      Prop("cameraLens") { (view: OlyPoseCameraView, lens: String) in view.options.lens = lens }
      Prop("cameraZoomFactor") { (view: OlyPoseCameraView, zoom: Double) in view.options.zoom = zoom }
      Prop("frameLimit") { (view: OlyPoseCameraView, limit: Int) in view.options.frameLimit = limit }
      Prop("poseModelVariant") { (view: OlyPoseCameraView, variant: String) in view.options.modelVariant = variant }
      Prop("poseModelAssetPath") { (view: OlyPoseCameraView, path: String?) in view.options.modelPath = path }
      OnViewDidUpdateProps { (view: OlyPoseCameraView) in view.applyChanges() }
    }
  }
}

File: android/src/main/java/expo/modules/olyposecamera/OlyPoseCameraModule.kt
package expo.modules.olyposecamera

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class OlyPoseCameraModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("OlyPoseCamera")
    View(OlyPoseCameraView::class) {
      Events("onCameraConfigured", "onLandmark", "onInferenceError")
      Prop("cameraFacing") { view: OlyPoseCameraView, facing: String -> view.options = view.options.copy(facing = facing) }
      Prop("cameraLens") { view: OlyPoseCameraView, lens: String -> view.options = view.options.copy(lens = lens) }
      Prop("cameraZoomFactor") { view: OlyPoseCameraView, zoom: Double -> view.options = view.options.copy(zoom = zoom) }
      Prop("frameLimit") { view: OlyPoseCameraView, limit: Int -> view.options = view.options.copy(frameLimit = limit) }
      Prop("poseModelVariant") { view: OlyPoseCameraView, variant: String -> view.options = view.options.copy(modelVariant = variant) }
      Prop("poseModelAssetPath") { view: OlyPoseCameraView, path: String? -> view.options = view.options.copy(modelPath = path) }
      OnViewDidUpdateProps { view: OlyPoseCameraView -> view.applyChanges() }
      OnViewDestroys { view: OlyPoseCameraView -> view.destroy() }
    }
  }
}
